src/Tizen.MachineLearning.Inference/Tizen.MachineLearning.Inference/TensorsInfo.cs

```diff
@@ -262,11 +262,127 @@ public TensorsData GetTensorsData()
             ret = Interop.Util.CreateTensorsData(_handle, out tensorsData_h);
             NNStreamer.CheckException(ret, "unable to create the tensorsData object");
 
-            retTensorData = TensorsData.CreateFromNativeHandle(tensorsData_h);
+            retTensorData = TensorsData.CreateFromNativeHandle(tensorsData_h, _handle, false);
 
             return retTensorData;
         }
 
+        /// <summary>
+        /// Gets the tensor name with given index.
+        /// </summary>
+        /// <param name="idx">The index of the tensor.</param>
+        /// <returns>The tensor name</returns>
+        /// <feature>http://tizen.org/feature/machine_learning.inference</feature>
+        /// <exception cref="ArgumentException">Thrown when the method failed due to an invalid parameter.</exception>
+        /// <exception cref="NotSupportedException">Thrown when the feature is not supported.</exception>
+        /// <since_tizen> 6 </since_tizen>
+        public string GetTensorName(int idx)
+        {
+            NNStreamer.CheckNNStreamerSupport();
+
+            CheckIndexBoundary(idx);
+            return _infoList[idx].Name;
+        }
+
+        /// <summary>
+        /// Gets the the hash code of this TensorsInfo object
+        /// </summary>
+        /// <returns>The hash code</returns>
+        /// <since_tizen> 7 </since_tizen>
+        public override int GetHashCode()
+        {
+            return this._infoList.GetHashCode();
+        }
+
+        /// <summary>
+        /// Compare TensorsInfo, which is its contents are the same or not.
+        /// </summary>
+        /// <param name="obj">Object to compare</param>
+        /// <returns>True if the given object is the same object or its contents are the same</returns>
+        /// <since_tizen> 7 </since_tizen>
+        public override bool Equals(object obj)
+        {
+            if (obj == null)
+                return false;
+
+            TensorsInfo cInfo = obj as TensorsInfo;
+            return this.Equals(cInfo);
+        }
+
+        /// <summary>
+        /// Compare TensorsInfo, which is its contents are the same or not.
+        /// </summary>
+        /// <param name="other">TensorsInfo instance to compare</param>
+        /// <returns>True if the given object is the same object or its contents are the same</returns>
+        /// <since_tizen> 7 </since_tizen>
+        public bool Equals(TensorsInfo other)
+        {
+            if (other == null)
+                return false;
+
+            if (this.GetHashCode() == other.GetHashCode())
+                return true;
+
+            if (this.Count != other.Count)
+                return false;
+
+            for (int i = 0; i < this.Count; ++i)
```

Comment:
Is the order of element also important to equals?
for example, 
`{ tensor1, tensor2 } != {tensor2, tensor1}` is it true?

Reply:
That's true. The order of `TensorInfo` in `TensorsInfo` obejct is important.